### PR TITLE
faster inv(::Rational)

### PR DESCRIPTION
**Pull request by stevengj**
A micro-optimization that I noticed while doing #22610.  About 3–4x faster than the generic `inv(::Number)` it was calling before.

### CODE REVIEW

**Comment by oscardssmith** [comment]
Does this work for 0?

**Comment by stevengj** [comment]
@oscardssmith, yes: `inv(0//1) === 1 // (0//1) === 1//0`

**Review by ararslan — APPROVED** [review]
:cool:

File: base/rational.jl [hunk 1]
```diff
@@ -255,6 +255,7 @@ function *(x::Rational, y::Rational)
 end
 /(x::Rational, y::Rational) = x//y
 /(x::Rational, y::Complex{<:Union{Integer,Rational}}) = x//y
+inv(x::Rational) = Rational(x.den,x.num)
```
Comment (tkelman) [inline base/rational.jl]: space after comma would be more readable

**Review by Sacha0 — APPROVED** [review]
lgtm modulo `@tkelman`'s minor comment! :)

**Pull request merged**